add llgo get parameter specification (#14)
seri037/llcppg: llpkgstore.md

llgo get module_path[@latest]
```

> The optional `latest` identifier is supported as a valid `cversion` or `module_version`. When `llgo get clib@latest`, The process first converts `clib` to `module_path`, which is then converted to `llgo get module_path@latest`. llgo will find the latest llpkg and pull it.

Reviewer: ```suggestion
The optional `latest` identifier is supported as a valid `cversion` or `module_version`. When `llgo get clib@latest`, `llgo get` will firstly convert `clib` to `module_path`, and then process it as `module_path@latest`. `llgo get` will find the latest llpkg and pull it.
```

llgo get module_path[@latest]
```

The optional `latest` identifier is supported as a valid `cversion` or `module_version`. When `llgo get clib@latest`, The process first converts `clib` to `module_path`, which is then converted to `llgo get module_path@latest`. llgo will find the latest llpkg and pull it.

Wrong usage:

```bash
llgo get clib@module_version
llgo get module_path@cversion
```

> The way `llgo get` handles the version number  after '@' is determined by the format of the content before '@'. For example, if it is a clib, it will be processed as `clib@cversion`; otherwise, it will be processed as `module_path@module_version`. This is to prevent the possibility that cversion and module_version might be the same.

Reviewer: ```suggestion
It's the format of the part before `@` that determines the how `llgo get` will handle the version; that is, `llgo get` will firstly check if it's a `clib`. If it is, the whole argument will be processed as `clib@cversion`; otherwise, it will be processed as `module_path@module_version`.
```

> **Details of `llgo get`**
>
>  1. `llgo` automatically resolves `clib@cversion` syntax into canonical `module_path@module_version` format.
>  2. Pull the go module by `go get`.
>  3. Check `llpkg.cfg` to determine if it's an llpkg. If it is:
>
>     - Run `conan install` to install binaries. `.pc` files for building will be stored in `${LLGOMODCACHE}`.
>     - Indicate the original `cversion` by adding a comment in `go.mod`. (We ignore indirect dependencies for now.)
>
>       ```
>       // go.mod 
>       require (
>             github.com/goplus/llpkg/cjson v1.1.0  // cjson 1.7.18
>       )
>       ```

## Listing clib version mapping
 

> This command `llgo list -m -versions clib`,  provides user the version mapping of an llpkg and is compatible with `go list`. 

Reviewer: ```suggestion
The command, `llgo list -m -versions clib`, provides user the version mapping of an llpkg, and is compatible with `go list`. 
```

*e.g.* `llgo list -m -versions cjson`:

```
module_path=github.com/goplus/llpkg/cjson 

> 1.3=v0.1.0
> 1.3=v1.0.1
> 1.3.1=v1.1.0

Reviewer: ```suggestion
1.3=v0.1.0
1.3=v0.1.1
1.3.1=v0.2.0
```

```

*e.g.* `llgo list -m -versions -json cjson`:

```json
{
  "CVersion": "1.7.18",
  "Mapping" : [{
      "C": "1.3",
      "Go": ["v0.1.0", "v0.1.1"]
  },
  {
      "C": "1.3.1",

>       "Go": ["v1.1.0"]

Reviewer: ```suggestion
      "Go": ["v0.2.0"]
```
